app/src/main/kotlin/com/simplemobiletools/smsmessenger/adapters/ThreadAdapter.kt

```diff
@@ -290,102 +290,170 @@ class ThreadAdapter(
+        val mimetype = attachment.mimetype
+        val uri = attachment.getUri()
+        parent.apply {
+            val imageView = layoutInflater.inflate(R.layout.item_attachment_image, null)
+            thread_mesage_attachments_holder.addView(imageView)
+
+            val placeholderDrawable = ColorDrawable(Color.TRANSPARENT)
+            val isTallImage = attachment.height > attachment.width
+            val transformation = if (isTallImage) CenterCrop() else FitCenter()
+            val options = RequestOptions()
+                .diskCacheStrategy(DiskCacheStrategy.RESOURCE)
+                .placeholder(placeholderDrawable)
+                .transform(transformation)
+
+            var builder = Glide.with(context)
+                .load(uri)
+                .transition(DrawableTransitionOptions.withCrossFade())
+                .apply(options)
+                .listener(object : RequestListener<Drawable> {
+                    override fun onLoadFailed(e: GlideException?, model: Any?, target: Target<Drawable>?, isFirstResource: Boolean): Boolean {
+                        thread_message_play_outline.beGone()
+                        thread_mesage_attachments_holder.removeView(imageView)
+                        return false
+                    }
+
+                    override fun onResourceReady(dr: Drawable?, a: Any?, t: Target<Drawable>?, d: DataSource?, i: Boolean) =
+                        false
+                })
+
+            builder = if (isTallImage) {
+                builder.override(attachment.width, attachment.width)
+            } else {
+                builder.override(attachment.width, attachment.height)
+            }
+
+            builder.into(imageView.attachment_image)
+            imageView.attachment_image.setOnClickListener {
+                if (actModeCallback.isSelectable) {
+                    holder.viewClicked(message)
+                } else {
+                    launchViewIntent(uri, mimetype, attachment.filename)
+                }
+            }
+            imageView.setOnLongClickListener {
+                holder.viewLongClicked()
+                true
+            }
+        }
+    }
 
-                                override fun onResourceReady(dr: Drawable?, a: Any?, t: Target<Drawable>?, d: DataSource?, i: Boolean) =
-                                    false
-                            })
+    private fun setupVCardView(holder: ViewHolder, parent: View, message: Message, attachment: Attachment) {
+        val uri = attachment.getUri()
+        parent.apply {
+            val vCardView = layoutInflater.inflate(R.layout.item_attachment_vcard, null).apply {
+                background.applyColorFilter(backgroundColor.getContrastColor())
+                vcard_title.setTextColor(textColor)
+                vcard_subtitle.setTextColor(textColor)
+                vcard_view_contact.setTextColor(context.getLinkTextColor())
```

Comment:
this seems to work weird.. If I use the default orange primary color, the "View contact details" is orange. If I change it in any way, it uses text color. Lets always use the proper primary color. Also make sure the whole UI works well with dark/light and system themes too.

Reply:
Looks good now. Tested with system light/dark and custom themes